src/io/aleph/dirigiste/Executor.java

```diff
@@ -134,26 +134,13 @@ public boolean shutdown() {
 
     private boolean _isShutdown = false;
 
-    private final AtomicReference<Stats.UniformLongReservoir> _queueLatencies =
-        new AtomicReference<Stats.UniformLongReservoir>(new Stats.UniformLongReservoir());
-
-    private final AtomicReference<Stats.UniformLongReservoir> _taskLatencies =
-        new AtomicReference<Stats.UniformLongReservoir>(new Stats.UniformLongReservoir());
```

Comment:
If you stop using AtomicReferences, we lose the `volatile` behavior of the reservoirs. Two threads in different cores may not know of each other's changes, overwrite them, or complete inconsistently.

Is this safe? I know the control loop is only on a single thread, and some reservoirs are only adjusted there, but the two latency-related reservoirs have `.add` called from fns like `acquire`/`release` (pools), and `execute` (executors), which all happen on other threads.

Reply:
Thanks for the review.
I need to think about it...

Reply:
We don't need to use an `AtomicReference` anymore as we'll never mutate the `RingBuffer` references once created.
However, I totally missed the fact that both the latency reservoirs are currently mutated on different threads.

I introduced an AtomicLongRingBuffer for both, which is actually what you pointed on your comment.